src/OrleansCodeGenerator/RoslynCodeGenerator.cs

```diff
@@ -72,22 +73,24 @@ public void AddGeneratedAssembly(string targetAssemblyName, GeneratedAssembly ge
         /// <summary>
         /// Generates code for all loaded assemblies and loads the output.
         /// </summary>
-        public void GenerateAndLoadForAllAssemblies()
+        public IReadOnlyList<GeneratedAssembly> GenerateAndLoadForAllAssemblies()
```

Comment:
I'm fine with converting this to a concrete collection. It shouldn't be a big deal - this isn't hot path stuff